librarian.js

```diff
@@ -292,12 +298,159 @@ program
+        fatalError("The IPA is missing critical information.");
+      }
+
+      const buildTime = new Date();
+      const folderName = buildTime.getTime();
+      const templatePath = prefs.working_directory + 'web/templates/manifest.plist';
+      const localManifestPath = prefs.working_directory + (prefs.assets_web ? 'asset_server' : 'web') + '/assets/b/' + folderName + '/local/manifest.plist';
+      const webManifestPath = prefs.working_directory + 'web/assets/b/' + folderName + '/web/manifest.plist';
+
+      try {
+        fs.copySync(templatePath, localManifestPath);
+        const manifest = fs.readFileSync(localManifestPath, 'utf8');
+        let editablePlist = plist.parse(manifest);
+        editablePlist.items[0].metadata["bundle-version"] = version;
+        editablePlist.items[0].metadata["bundle-identifier"] = bundleIdentifier;
+        editablePlist.items[0].metadata.title = appName;
+        editablePlist.items[0].assets[0].url = fileUploadResult;
+        fs.writeFileSync(localManifestPath, JEYLL_FRONT_MATTER_CHARACTER + plist.build(editablePlist));
+        if (options.public && !prefs.assets_web) {
+          fs.copySync(templatePath, webManifestPath);
+          editablePlist.items[0].assets[0].url = fileUploadResult;
+          fs.writeFileSync(webManifestPath, JEYLL_FRONT_MATTER_CHARACTER + plist.build(editablePlist));
+        }
+      } catch (error) {
+        fatalError(error);
+      }
+      // s3_url --> new variable for pointing the resource path available in s3.
+      buildInfo = {
+        "version": version,
+        "buildNumber": build,
+        "bundle": bundleIdentifier,
+        "folderPath": folderName,
+        "s3_url": fileUploadResult,
+        "date": buildTime.toISOString()
+      };
+    } else {
+      const appName = metadata.originalFileName;
+      const buildTime = new Date();
+      const folderName = buildTime.getTime();
+      const apkPath = fileUploadResult;
+
+      if (bundleIdentifier === undefined || appName === undefined || version === undefined || build === undefined) {
+        fatalError("The APK is missing critical information.");
+      }
+
+      // s3_url --> new variable for pointing the resource path available in s3.
+      buildInfo = {
+        "version": version,
+        "buildNumber": build,
+        "bundle": bundleIdentifier,
+        "folderPath": folderName,
+        "fileName": appName,
+        "s3_url": fileUploadResult,
+        "date": buildTime.toISOString()
+      };
+    }
+
+    buildInfo.notes = options.notes ? options.notes : "";
+    buildInfo.branch = options.branch ? options.branch : "";
+    buildInfo.public = !!options.public;
+    buildInfo.platform = platform;
+    await addBuild(preferences, buildInfo);
+    printHeader("Build Added Successfully!");
     await checkForUpdate(preferences);
+    if (options.delete_apk.toString() === "1") { fs.unlink(pathToFile, () => { }) }
```

Comment:
What's this?

Reply:
to remove the local build file, if the user wants.